src/SpectatorView.Native/SpectatorView.Compositor/Compositor/AzureKinectFrameProvider.cpp

```diff
@@ -210,25 +215,41 @@ uint8_t* AzureKinectFrameProvider::GetBodyIndexBuffer(k4a_capture_t capture)
     return bodyIndexBuffer;
 }
 
-void AzureKinectFrameProvider::SetBodyDepthBuffer(uint16_t* bodyDepthBuffer, uint16_t* depthBuffer, uint8_t* bodyIndexBuffer, int bufferSize)
+void AzureKinectFrameProvider::SetBodyMaskBuffer(uint16_t* bodyMaskBuffer, uint8_t* bodyIndexBuffer, int bufferSize)
 {
     int bufferIndex = 0;
 
-    //Copy depth values only if bodyID is not K4ABT_BODY_INDEX_MAP_BACKGROUND
     while (bufferIndex < bufferSize)
     {
         if (bodyIndexBuffer[bufferIndex] != K4ABT_BODY_INDEX_MAP_BACKGROUND)
         {
-            bodyDepthBuffer[bufferIndex] = depthBuffer[bufferIndex];
+            // Using 1000 to ensure value is not truncated to 0 by depth to color transformation
+            bodyMaskBuffer[bufferIndex] = 1000;
         }
         else
         {
-            bodyDepthBuffer[bufferIndex] = 0;
+            bodyMaskBuffer[bufferIndex] = 0;
+        }
+
+        bufferIndex++;
+    }
+}
+
+void AzureKinectFrameProvider::SetTransformedBodyMaskBuffer(uint16_t* transformedBodyMaskBuffer, int bufferSize)
+{
+    int bufferIndex = 0;
+
+    while (bufferIndex < bufferSize)
```

Comment:
>while (bufferIndex < bufferSize) [](start = 4, length = 32)

I would find a for loop more intuitive to read here than a while loop with an increment at the end

Reply:
Agreed